### ISSUE TITLE
new blocks always introduce new scope

### ISSUE THREAD

**Issue by vitreo12** [vitreo12/omni#97]
Here `value` are two different variable declarations (due to `declaredInScope`)... This forces the user to declare any re-usable variable in `init`.

```nim
sample:
    value = 0.0
    for i in 0..10:
        value = value + i
```

```nim
init:
    value sig

sample:
    value = 0.0
    for i in 0..10:
        value = value + i
```

**Comment by vitreo12** [comment]
Things to solve regarding this:

`declared` would also not allow to name variables after names of functions OR the name of the top level module that's compiled (the other ones are imported with the `_module` appended thanks to `require`)

```
init:
    gamma = 0.5
```

`fourses.nim`:

```
init:
    fourses = 0.0
```

**Comment by vitreo12** [comment]
The solution for the second case would be to know the name of the compiled file at compile time.

**Comment by vitreo12** [comment]
fixed in #99 

Both cases work now. Simply used a `seq[string]` to keep track of all variables declared in a block
